File: backend/app/schemas/logistics.py
# app/schemas/logistics.py
from datetime import datetime
from pydantic import BaseModel
from app.models.logistics import LogisticsStatus


class LogisticsEvent(BaseModel):
    """物流事件"""
    time: str
    description: str
    location: str | None = None


class LogisticsResponse(BaseModel):
    id: int
    order_id: int
    tracking_number: str | None = None
    carrier: str | None = None
    status: LogisticsStatus
    events: list[LogisticsEvent] = []
    estimated_delivery: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SimulateLogisticsRequest(BaseModel):
    """模拟物流更新请求"""
    tracking_number: str = "SF"  # 默认顺丰
    carrier: str = "顺丰速运"

File: frontend/src/types/logistics.ts
// src/types/logistics.ts
export interface LogisticsEvent {
  time: string;
  description: string;
  location: string | null;
}

export interface Logistics {
  id: number;
  order_id: number;
  tracking_number: string | null;
  carrier: string | null;
  status: 'pending' | 'shipped' | 'in_transit' | 'delivering' | 'delivered' | 'exception';
  events: LogisticsEvent[];
  estimated_delivery: string | null;
  created_at: string;
  updated_at: string;
}

// 物流状态中文映射
export const LogisticsStatusText: Record<string, string> = {
  pending: '待发货',
  shipped: '已发货',
  in_transit: '运输中',
  delivering: '派送中',
  delivered: '已签收',
  exception: '异常',
};

// 物流状态对应的颜色
export const LogisticsStatusColor: Record<string, string> = {
  pending: '#6c757d',
  shipped: '#0d6efd',
  in_transit: '#ffc107',
  delivering: '#fd7e14',
  delivered: '#198754',
  exception: '#dc3545',
};
